cnd/src/swap_protocols/rfc003/ethereum/htlc_events.rs

```diff
@@ -50,21 +50,13 @@ impl HtlcDeployed<Ethereum, asset::Ether> for Cache<Web3Connector> {
         start_of_swap: NaiveDateTime,
     ) -> anyhow::Result<Deployed<Ethereum>> {
         let connector = self.clone();
-        let pattern = TransactionPattern {
-            from_address: None,
-            to_address: None,
-            is_contract_creation: Some(true),
-            transaction_data: Some(htlc_params.bytecode()),
-            transaction_data_length: None,
-            events: None,
-        };
-        let TransactionAndReceipt { transaction, .. } =
-            matching_transaction(connector, pattern, start_of_swap)
-                .await
-                .context("failed to find transaction for htlc deployment")?;
+        let transaction =
+            matching_create_contract_transaction(connector, start_of_swap, htlc_params.bytecode())
+                .await?;
+        let location = calculate_contract_address_from_deployment_transaction(&transaction);
```

Comment:
The contract address is actually in the receipt!

`matching_create_contract_transaction` could return us the location from the receipt, then we would not need to calculate it and could throw away a whole bunch of code (and some dependencies!) :)